arangod/GeneralServer/RestHandler.cpp

```diff
@@ -781,3 +781,9 @@ void RestHandler::resetResponse(rest::ResponseCode code) {
   TRI_ASSERT(_response != nullptr);
   _response->reset(code);
 }
+
+futures::Future<futures::Unit> RestHandler::executeAsync() {
+  THROW_ARANGO_EXCEPTION(TRI_ERROR_NOT_IMPLEMENTED);
+}
+
+RestStatus RestHandler::execute() { return waitForFuture(executeAsync()); }
```

Comment:
I am a bit confused about how this can actually work.
This PR adds `executeAsync()` as a new virtual method, but it is only overridden in a single RestHandler.

And the default implementation of RestHandler::execute() now changed to call executeAsync(), which be default will call the base class executeAsync(), which will throw an exception.

Reply:
Either implement `execute` or `executeAsync`. If you implement both `execute` will be called.